src/main/java/com/example/lures/LureService.java

```diff
@@ -35,4 +35,22 @@ public Lure insert(String product, String company, double size, double weight) {
         lureMapper.insert(lure);
         return lure;
     }
+
+    public Lure update(Integer id, String product, String company, double size, double weight) {
+        Optional<Lure> optionalLure = this.lureMapper.findById(id);
+        Lure lure = optionalLure.orElseThrow(() -> new LureNotFoundException("Lure not found"));
+
+        Optional<Lure> existingLure = lureMapper.findByLure(product);
+        if (!existingLure.isEmpty() && !existingLure.stream().anyMatch(l -> l.getId().equals(id))) {

```

Comment:
否定形（ ! ）よりも肯定的な名前の関数を使うほうがよみやすいです。
```suggestion
        if (existingLure.isPresent() && existingLure.stream().noneMatch(l -> l.getId().equals(id))) {
```
IntelliJも警告出しているはず。
<img width="845" alt="スクリーンショット 2024-04-18 11 54 57" src="https://github.com/MasatoKihara25/last-assignment/assets/62045457/4b79f8a8-f733-4850-9bf3-690e9377ea7e">

Reply:
肯定文の名前の関数の方がよみやすいので、そちらで修正してマージします。